src/dbnode/persist/fs/migration/migration.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package migration
+
+import (
+	"github.com/m3db/m3/src/dbnode/namespace"
+	"github.com/m3db/m3/src/dbnode/persist"
+	"github.com/m3db/m3/src/dbnode/persist/fs"
+	"github.com/m3db/m3/src/dbnode/storage"
+	xtime "github.com/m3db/m3/src/x/time"
+)
+
+// Migration interface is implemented by tasks that wish to perform a data migration
+// on a fileset. Typically involves updating files in a fileset that were created by
+// a previous version of the database client
+type Migration interface {
+
+	// Run is the set of steps to successfully complete a migration
+	Run() error
+}
+
+var _ Migration = &ToVersion1_1{}
+
+// ToVersion1_1 is an object responsible for migrating a fileset to version 1.1
+type ToVersion1_1 struct {
+	newMergerFn fs.NewMergerFn
+	infoFile    fs.ReadInfoFileResult
+	shard       uint32
+	namespace   namespace.Metadata
+	opts        storage.Options
+}
+
+func NewToVersion1_1(infoFile fs.ReadInfoFileResult, shard uint32, namespace namespace.Metadata, opts storage.Options) Migration {
```

Comment:
nit: should we merge these into some sort of `MigrationOptions` struct w/ some validation and defaults (in some places)?